### PR TITLE
docs(nx): add in next infrastructure

### PR DESCRIPTION
**Pull request by MetecanAydin**

### CODE REVIEW

File: docs/nx/next/infra.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+### @ensono-stacks/next:infrastructure
+
+<details>
+<summary>Configure Infrastructure for your Next project</summary>
+
+The infrastructure generator will provide all the necessary tools and setup ready to configure your pipeline.
```
Comment (cloudratha) [inline docs/nx/next/infra.md:6]: Technically not configuring a pipeline at all here. More infrastructure required to host your application in a Kubernetes Cluster.

File: docs/nx/next/infra.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+- installs following dev dependencies
+
+```
+@nx-tools/nx-container
+@nx-tools/container-metadata
+@jscutlery/semver
+```
+
+- Adds `stacks` object to nx.json which is populated via interactive and pulled in various stages of the infra journey
```
Comment (cloudratha) [inline docs/nx/next/infra.md:65]: It shouldn't add anything realted to nx.json. but reads it to know how to scaffold the infra as code values. Its a requirement

**Review by cloudratha — CHANGES_REQUESTED** [review]
@johnmckenna please can you review this too.

File: docs/nx/next/infra.md (hunk 2 above)
Comment (CerdenAmido) [reply docs/nx/next/infra.md:65]: I had this on another plugin - I believe the first generator you run after `create-stacks-workspace` may add some fields to nx.json depending on what's already there - worth checking if this file always gets changed depending on settings etc

**Review by johnmckenna — CHANGES_REQUESTED** [review]

**Review by johnmckenna — APPROVED** [review]

**Pull request merged**